internal/archive/archive_test.go

```diff
@@ -331,50 +331,47 @@ func (s *httpSuite) TestArchiveLabels(c *C) {
 		}
 	}
 
-	s.prepareArchive("jammy", "22.04", "amd64", []string{"main", "universe"})
+	tests := []struct {
+		summary string
+		label   string
+		err     string
+	}{{
+		summary: "No labels",
```

Comment:
Because the default label is set to `Ubuntu` in the code.

```
	// Parse the appropriate section for the type of archive.
	label := "Ubuntu"
	if index.archive.options.Pro != "" {
		label = proArchiveInfo[index.archive.options.Pro].Label
	}
```

With this test case, we are testing the default behaviour.

Reply:
No wait, this is not testing that. The label here refers to `testarchive.Release`, see:
```go
func (s *httpSuite) prepareArchiveAdjustRelease(suite, version, arch string, components []string, adjustRelease func(*testarchive.Release)) *testarchive.Release {
	release := &testarchive.Release{
		Suite:   suite,
		Version: version,
		Label:   "Ubuntu",
		PrivKey: s.privKey,
	}
	...
	if adjustRelease != nil {
		adjustRelease(release) // set release.Label = <whatever>
	}
```

This test is checking the default implementation of `testarchive.Release` which is not the thing that we want. We should either have a test with "Ubuntu" explicitly or that uses the default (and say no label which is not true, it is the default). I think it is better to have the explicit "Ubuntu" because the fact that it is the default is just for convenience in the rest of the tests.

Reply:
Also, correct me if I am wrong but I don't think we are using the label from archive.Options at all, so I wonder if we should remove it or use it.

Reply:
The `label` is useful. It passes the archive name, as defined in chisel.yaml. Using it https://github.com/letFunny/chisel/pull/13/commits/47c92d9f4a5e59da7a244a0c15cf3292c9c26a16 actually!

Reply:
Removed the useless test: https://github.com/letFunny/chisel/pull/13/commits/7ff02acae7104ce7459de7773a4abebd72d81c2c.

Reply:
Label is indeed useful, thanks for that!